swift/extractor/infra/SwiftDispatcher.h

```diff
@@ -142,9 +129,10 @@ class SwiftDispatcher {
   // This method gives a TRAP label for already emitted AST node.
   // If the AST node was not emitted yet, then the emission is dispatched to a corresponding
   // visitor (see `visit(T *)` methods below).
-  template <typename E, std::enable_if_t<IsFetchable<E>>* = nullptr>
-  TrapLabelOf<E> fetchLabel(const E& e, swift::Type type = {}) {
-    if constexpr (std::is_constructible_v<bool, const E&>) {
+  template <typename E>
+  requires std::constructible_from<Handle, E> TrapLabelOf<E> fetchLabel(const E& e,
```

Comment:
> Is the Clang-Format update expected to fix this?
yes, see https://github.com/github/codeql/pull/13991/commits/970b3d06beb3ff5acd4ba753d15d898ef7a37397#diff-49d21143bb333c21c7dea231e8120ddd95c32572d43cc21d5c01e9cbe66ffc98
However I can put a couple of `// clang-format off` for the moment, as indeed that is not really readable

> (I guess we have to use the explicit requires syntax to support the pointer overload?)

Yes and no. The way constraints work, in order to just be able to compile only one of the two is required. As opposed to `std::enable_if`, you don't need disjoint requirements to sort out overload resolution, because concepts subsume. In particular, any requires clause will take precedence over no requires clause in overload resolution, without ambiguity.

However, constraints also help with compiler errors. For example calling `fetchLabel` on a `const swift::Argument&` to test it out now yields something containing:
```
./swift/extractor/infra/SwiftDispatcher.h:168:18: note: candidate template ignored: constraints not satisfied [with E = swift::Argument]
  TrapLabelOf<E> fetchLabel(const E& e) {
                 ^
./swift/extractor/infra/SwiftDispatcher.h:167:19: note: because 'std::constructible_from<codeql::SwiftDispatcher::Handle, swift::Argument *>' evaluated to false
    requires std::constructible_from<Handle, E*>
```
Without the constraint the error message is quite more convoluted. So I think it's better to keep both, even if both are not required to compile.

Reply:
> However, constraints also help with compiler errors.

Agreed. I meant more whether we could do something like `const IsFetchable auto& e` for one and `const IsFetchable auto* e` for the other (but obviously not that because that's still checking `IsFetchable<T>` in both cases).